modules/instrument_builder/js/react.app.js

```diff
@@ -0,0 +1,485 @@
+				instrumentName: value
+			};
+		});
+	},
+	render: function () {
+		var value = this.state.fileName;
+		return (
+			React.createElement(TabPane, {Title: "Save Instrument", 
+                TabId: this.props.TabId}, 
+                	React.createElement("div", {className: "form-group"}, 
+                		React.createElement("div", {className: "col-xs-12"}, 
+			                React.createElement("label", {className: "col-sm-2 control-label"}, "Filename: "), 
+			                React.createElement("div", {className: "col-sm-4"}, 
+			                    React.createElement("input", {className: "form-control", 
+			                    	   type: "text", id: "filename", 
+			                    	   value: value, 
+			                    	   onChange: this.onChangeFile}
+			                   	)
+			                )
+			            ), 
+			            React.createElement("br", null), React.createElement("br", null), 
+			            React.createElement("div", {className: "col-xs-12"}, 
+			                React.createElement("label", {className: "col-sm-2 control-label"}, "Instrument Name: "), 
+			                React.createElement("div", {className: "col-sm-4"}, 
+			                    React.createElement("input", {className: "form-control", 
+			                    	   type: "text", id: "longname", 
+			                    	   value: this.state.instrumentName, 
+			                    	   onChange: this.onChangeInst}
+			                    )
+			                )
+			            ), 
+			            React.createElement("br", null), React.createElement("br", null), 
+			            React.createElement("br", {className: "visible-xs"}), 
+			            React.createElement("br", {className: "visible-xs"}), 
+			            React.createElement("br", {className: "visible-xs"}), 
+			            React.createElement("div", {className: "col-xs-12"}, 
+			            	React.createElement("div", {className: "col-xs-6 col-sm-2 col-sm-offset-2"}, 
+			                	React.createElement("input", {className: "btn btn-primary col-xs-12", type: "button", onclick: "Instrument.validate()", value: "Validate"})
+			                ), 
+			                React.createElement("div", {className: "col-xs-6 col-sm-2"}, 
+			                	React.createElement("input", {className: "btn btn-primary col-xs-12", 
+			                		   type: "submit", value: "Save", 
+			                		   onClick: this.props.save}
+			                	)
+			                )
+			            )
+		            )
+			)
+		);
+	}
+});
+
+DisplayElements = React.createClass({displayName: "DisplayElements",
+	getPlaceholder: function() {
+	    if (!this.placeholder) {
+	      var tr = document.createElement('tr');
+	      tr.className = "placeholder";
+	      var td = document.createElement('td');
+	      td.colSpan = 2;
+	      td.appendChild(document.createTextNode("Drop here"));
+	      tr.appendChild(td);
+	      this.placeholder = tr;
+	    }
+	    return this.placeholder;
```

Comment:
Why is this done with direct DOM manipulation instead of JSX since it's in a react component?